File: typescript/1.two-sum.ts
/*
 * @lc app=leetcode id=1 lang=typescript
 *
 * [1] Two Sum
 */

// @lc code=start
function twoSum(nums: number[], target: number): number[] {
    const numMap = new Map<number,number>();

    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];

        if (numMap.has(complement)) {
            return [numMap.get(complement)!, i]
        }
        numMap.set(nums[i], i);
    }
    return []
    
};
// @lc code=end

File: go/1.two-sum.go
/*
 * @lc app=leetcode id=1 lang=golang
 *
 * [1] Two Sum
 */

// @lc code=start

package main

// using hashmap
func twoSum(nums []int, target int) []int {
	numMap := make(map[int]int) //value -> index

	for i, num := range nums {
		complement := target - num
		if j, exists := numMap[complement]; exists {
			return []int{j, i}
		}
		numMap[num] = i
	}
	return []int{}
}

// @lc code=end
